### ISSUE TITLE
My model look like as multilabel  classifier instead of binary classifier 

### ISSUE THREAD

**Issue by talhaanwarch** [amaiya/ktrain#298]
I am trying to do binary classifier, but the output looks like multi-label classification
Here are the steps
```
import ktrain
from ktrain import text
MODEL_NAME = 'bert-base-multilingual-cased'
t = text.Transformer(MODEL_NAME, maxlen=512,classes=['real','fake'])

trn = t.preprocess_train(train_text, train_labels)
tst = t.preprocess_test(test_text, test_labels)
model = t.get_classifier()

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
learner = ktrain.get_learner(model, train_data=trn, val_data=tst, batch_size=8)
learner.fit_onecycle(0.00001, 5) 

predictor = ktrain.get_predictor(learner.model, preproc=t)
y_pred=predictor.predict(test_text,return_proba=True)

```
The output looks like
```
[0.65020806, 0.5859068 ],
[0.6465919 , 0.63063735],
 [0.63927495, 0.62223727],
 [0.6334275 , 0.6333737 ]
```

Colab link:
https://colab.research.google.com/drive/1Y_m4Su2ER5zkYiraAViLZcEOgVy4_NJv?usp=sharing
So, the last layer is not sigmoid, but in binary classification, the last layer should be sigmoid. How can I resolve it.

**Comment by amaiya** [comment]
Hello,

Remove this line:
```python
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
```

**ktrain** automatically detects binary/multi-class/multi-label problems from the data and configures the model appropriately, as demonstrated in the tutorial and example notebooks.

In this case, the dataset is preprocessed as 1-hot encoded with two columns for the label (i.e., binary classification) and correctly configures the model to use `categorical_crossentropy`, but you have changed it to `binary_crossentropy`.    You can use `categorical_crossentropy` for binary classification problems if the labels are 1-hot encoded (e.g.., `[1,0]` -> 0 and `[0,1]` -> 1), which is the default in **ktrain**.
